pkg/kapp/app/recorded_app.go

```diff
@@ -146,13 +146,13 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
 	}
 
 	if foundNonMigratedApp {
-		if a.isMigrationEnabled() {
+		if a.isMigrationEnabled() && !isDiffRun {
```

Comment:
It will go to update only and there is no change that will happen when it's diff-run. But yeah returning nil earlier itself may be better. I will make the change in a separate PR.

Reply:
I think It will be fine as it will go ahead and call `updateApp()` and as the DiffRun is True the update will tale place with DryRun and nothing will change on the cluster. 

I believe [here](https://github.com/vmware-tanzu/carvel-kapp/blob/4a6a3a60851b1425cbdb4acb40566d4e7e788d72/pkg/kapp/app/recorded_app.go#L259) we have to take care that to `return nil` as both will gonna make changes on the cluster